[Edit] Update terminal.md documentation (#3861)
Codecademy/docs: content/command-line/concepts/terminal/terminal.md

The terminal interacts with a program often referred to as the *shell*, which is the command-line interface that exposes human input from the terminal to the operating system. Unlike the terminal, the shell is what actually processes the commands, whereas the terminal is just the text-based interface to interact with the shell. Some shells come as standalone terminals as well, such as Powershell or Command Prompt, however, some, such as the Unix BASH (Bourne Again SHell) require a separate terminal program.

## Examples of terminals

Terminals come with different features and interfaces based on the system:
- **macOS** has a native app called "Terminal" and also supports iTerm2, a popular alternative.
- **Windows** users typically work with:
  - Command Prompt (CMD), a legacy terminal based on MS-DOS that is available as a shell.
  - PowerShell, which is available as a standalone terminal or as a shell that offers advanced scripting capabilities.
  - Windows Terminal, the modern, multi-tabbed application supporting Command Prompt, PowerShell, and WSL (Windows Subsystem for Linux) shells.

>     - WSL, also called Windows Subsystem for Linux, is an emulation layer that allows you to use Unix-based shells such as BASH or ZSH. It can be run within other terminal emulators or as a standalone program. You can read more about WSL [here](https://learn.microsoft.com/en-us/windows/wsl/about)

Reviewer: ```suggestion
    - WSL, also called Windows Subsystem for Linux, is an emulation layer that allows the use of Unix-based shells such as BASH or ZSH. It can be run within other terminal emulators or as a standalone program. Read more about WSL [here](https://learn.microsoft.com/en-us/windows/wsl/about).
```

Author: Thanks for the change. I tried to avoid doing that, but I must've missed it on the WSL part (I wrote it a bit later, oopsies.)

A **terminal** is a tool for interacting with a computer using text-based commands. In the past, terminals were hardware-based systems that printed out typed commands to a dedicated hardware, such as a screen or paper spool. Today, terminals have been widely replaced by **Terminal Emulators**, which are GUI apps that can come with additional functionality, such as tabbing interfaces and splitting views. The words are used interchangeably in today's development environments. Terminals can serve various functions, like running scripts or automating tasks with shell scripts. 

The terminal interacts with a program often referred to as the *shell*, which is the command-line interface that exposes human input from the terminal to the operating system. Unlike the terminal, the shell is what actually processes the commands, whereas the terminal is just the text-based interface to interact with the shell. Some shells come as standalone terminals as well, such as Powershell or Command Prompt, however, some, such as the Unix BASH (Bourne Again SHell) require a separate terminal program.

## Examples of terminals

Terminals come with different features and interfaces based on the system:

> 

Reviewer 2: ```suggestion
```
